lib/file_store/s3_store.rb

```diff
@@ -94,12 +94,10 @@ def store_file(file, path, opts = {})
           opts[:content_type].presence || MiniMime.lookup_by_filename(filename)&.content_type,
       }
 
-      # add a "content disposition: attachment" header with the original
-      # filename for everything but safe images (not SVG). audio and video will
-      # still stream correctly in HTML players, and when a direct link is
-      # provided to any file but an image it will download correctly in the
-      # browser.
-      if !FileHelper.is_inline_image?(filename)
+      # Only add a "content disposition: attachment" header for svgs
+      # see https://github.com/discourse/discourse/commit/31e31ef44973dc4daaee2f010d71588ea5873b53.
+      # Adding this header for all files would break the ability to view attachments in the browser
+      if FileHelper.is_svg?(filename)
```

Comment:
This change also needs to be made in `to_s3_migration` I think:

https://github.com/discourse/discourse/blob/3260654505357fad33e944bd6d3e8ae3e14590de/lib/file_store/to_s3_migration.rb#L258-L264